xknxproject/models/models.py

```diff
@@ -47,6 +54,27 @@ def __repr__(self) -> str:
         )
 
 
+@dataclass
+class XMLGroupRange:
+    """Class that represents a group range."""
+
+    name: str
+    range_start: int
+    range_end: int
+    group_addresses: list[int]
+    children: list[XMLGroupRange]
+    comment: str
+
+    def str_address(self) -> str:
+        """Generate a string representation for the range (same as in in ETS)."""
+        main = (self.range_start & 0b1111100000000000) >> 11
+        middle = (self.range_start & 0b11100000000) >> 8
+        if (self.range_end - self.range_start) > (2**16 / 32 - 3):
```

Comment:
that's equivalent to `> 2045.0`. Why is that?
3-level subgroups have max. 255
2-level subgroups have max. 2047

... if you decide to use extra dataclasses for Main and Middle it wouldn't matter 😉

Reply:
Modified the approach here significantly -- so hopefully it fits better now ...